libp2p/Host.cpp

```diff
@@ -739,27 +725,18 @@ void Host::run(boost::system::error_code const&)
         }
     }
 
+    if (!m_run)
+        return;
+
     auto runcb = [this](boost::system::error_code const& error) { run(error); };
-    m_timer->expires_from_now(boost::posix_time::milliseconds(c_timerInterval));
-    m_timer->async_wait(runcb);
+    m_timer.expires_from_now(boost::posix_time::milliseconds(c_timerInterval));
```

Comment:
Just a question: Can you use `std::chrono` instead of `boost` for milliseconds?

Reply:
Unfortunately no, expires_from_now requires boost time.